src/producer/base_producer.rs

```diff
@@ -362,19 +360,32 @@ where
     /// Regular calls to `poll` are required to process the events and execute
     /// the message delivery callbacks.
     pub fn poll<T: Into<Timeout>>(&self, timeout: T) {
-        let event = self.client().poll_event(&self.queue, timeout.into());
-        if let EventPollResult::Event(ev) = event {
-            let evtype = unsafe { rdsys::rd_kafka_event_type(ev.ptr()) };
-            match evtype {
-                rdsys::RD_KAFKA_EVENT_DR => self.handle_delivery_report_event(ev),
-                _ => {
-                    let evname = unsafe {
-                        let evname = rdsys::rd_kafka_event_name(ev.ptr());
-                        CStr::from_ptr(evname).to_string_lossy()
-                    };
-                    warn!("Ignored event '{}' on base producer poll", evname);
+        let mut remaining = if let Timeout::After(dur) = timeout.into() {
+            dur
+        } else {
+            Duration::MAX
+        };
+        let mut attempt = 0;
```

Comment:
if `attempt` is 0 then we won't run even single iteration of the loop below.

Reply:
Fixed in 204dbe2f.

Reply:
Do we need `attempt` at all? It starts with 0 as it only incremented (so always >= 0). In while loop condition `attempt >= 0` so it'll be always true.

Reply:
I added that to ensure that if someone passed `Timeout::After(Duration::ZERO)` for a non-blocking call, we would always call the underlying `librdkafka` method at least once.

Reply:
not really as `&&` is used in the condition.

Reply:
Okay; I believe I fixed the logic error in 1957d2f01249ad96c887c4f365f6cd26ef1394a8.